airbyte-integrations/connectors/source-zuora/source_zuora/source.py

```diff
@@ -0,0 +1,427 @@
+            "list": type_array,
+            "array": type_array,
+            "boolean": type_bool,
+            "bool": type_bool,
+        }
+
+        json_schema = {}
+        for field in response:
+            json_type = type_mapping.get(field.get("Type"), type_string)
+            json_schema[field.get("Column")] = {"type": json_type}
+
+        return [json_schema]
+
+
+class ZuoraSubmitJob(ZuoraStream):
+
+    http_method = "POST"
+
+    def __init__(self, query: str, config: Dict):
+        super().__init__(config)
+        self.query = query
+
+    def path(self, **kwargs) -> str:
+        return "/query/jobs"
+
+    def request_body_json(self, **kwargs) -> Optional[Mapping]:
+        params = self._request_params(stream_state=None)
+        params["query"] = self.query
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> List[str]:
+
+        """
+        Response example:
+        {'data':
+            {
+                'id': 'c6f25f91-5357-4fec-a00d-9009cc1ae856',
+                'query': 'DESCRIBE account',
+                'useIndexJoin': False,
+                'sourceData': 'LIVE',
+                'queryStatus': 'accepted',
+                'remainingRetries': 3,
+                'retries': 3,
+                'updatedOn': '2021-07-26T15:33:48.287Z',
+                'createdBy': '84f78cea-8a5b-4332-933f-27439fe3b87b'
+            }
+        }
+        """
+        return [response.json()["data"]["id"]]
+
+
+class ZuoraJobStatusCheck(ZuoraStream):
+    def __init__(self, job_id: str, config: Dict):
+        super().__init__(config)
+        self.job_id = job_id
+
+    def path(self, **kwargs) -> str:
+        return f"/query/jobs/{self.job_id}"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> List[str]:
+        return [response.json()["data"]["dataFile"]]
+
+    def _send_request(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> requests.Response:
+
+        """
+        {'data':
+            {
+                'id': 'c6f25f91-5357-4fec-a00d-9009cc1ae856',
+                'query': 'DESCRIBE account',
+                'useIndexJoin': False,
+                'sourceData': 'LIVE',
+                'queryStatus': 'completed',
+                'dataFile': 'https://owl-auw2-sbx01-query-result.s3.us-west-2.amazonaws.com/c6f25f91-5357-4fec-a00d-9009cc1ae856_2779514650704989.jsonl?....',
+                'outputRows': 53,
+                'processingTime': 516,
+                'remainingRetries': 3,
+                'retries': 3,
+                'updatedOn': '2021-07-26T15:33:48.803Z',
+                'createdBy': '84f78cea-8a5b-4332-933f-27439fe3b87b'
+            }
+        }
+        """
+
+        # Define the job error statuses
+        errors = ["failed", "canceled", "aborted"]
+        # Error msg: the cursor_field cannot be resolved
+        cursor_error = "Column 'updateddate' cannot be resolved"
```

Comment:
current value of cursor_field can be different. I propose to generate this error message dynamically 
`"Column 'updateddate' cannot be resolved" => f"Column '{self.cursor_field}' cannot be resolved"`

Reply:
Sure, thank you.